python/google/protobuf/internal/message_test.py

```diff
@@ -1638,6 +1638,12 @@ def testMapByteSize(self):
     msg.map_int32_foreign_message[19].c = 128
     self.assertEqual(msg.ByteSize(), size + 1)
 
+  def testMapFieldConstruction(self):
+    msg1 = map_unittest_pb2.TestMap()
+    msg1.map_string_foreign_message['test'].c = 42
+    msg2 = map_unittest_pb2.TestMap(map_string_foreign_message=msg1.map_string_foreign_message)
+    self.assertEqual(42, msg2.map_string_foreign_message['test'].c)
```

Comment:
Does this test work well? Is it a behavior diffs between pure python and c extension?

Reply:
Yes, this is a minimal equivalent to the case we found when trying to migrate our codebase to the C++ implementation - it was fine in the pure-Python version but the field was not initialised correctly with the C++ version.